tests/integration/userCompanies.test.js

```diff
@@ -0,0 +1,316 @@
+        method: 'PUT',
+        url: `/usercompanies/${userCompanyId}`,
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(403);
+    });
+
+    it('should return a 403 if detachment date is before first trainee\'s addition in course history', async () => {
+      const userCompanyId = userCompanies[0]._id.toHexString();
+      const payload = { endDate: '2022-08-17T10:00:00.000Z' };
+
+      const res = await app.inject({
+        method: 'PUT',
+        url: `/usercompanies/${userCompanyId}`,
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(403);
+      expect(res.result.message).toBe('Vous ne pouvez pas détacher cette personne avant le 05/09/2022.');
+    });
+  });
+
+  describe('TRAINING_ORGANISATION_MANAGER', () => {
+    beforeEach(async () => {
+      authToken = await getTokenByCredentials(usersSeedList[3].local);
+    });
+
+    it('should detach user Company', async () => {
+      const userCompanyId = userCompanies[5]._id.toHexString();
+      const payload = { endDate: '2022-12-01T22:59:59.999Z' };
+
+      const res = await app.inject({
+        method: 'PUT',
+        url: `/usercompanies/${userCompanyId}`,
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(200);
+
+      const userCount = await UserCompany.countDocuments({ _id: userCompanyId, endDate: '2022-12-01T22:59:59.999Z' });
+      expect(userCount).toEqual(1);
+    });
+
+    it('should return a 400 if endDate is not defined in payload', async () => {
+      const userCompanyId = userCompanies[5]._id.toHexString();
+      const payload = { endDate: '' };
+
+      const res = await app.inject({
+        method: 'PUT',
+        url: `/usercompanies/${userCompanyId}`,
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(400);
+    });
+
+    it('should return a 403 if user has a role', async () => {
+      const userCompanyId = userCompanies[7]._id;
+      const payload = { endDate: '2022-12-25T22:59:59.999Z' };
+
+      const res = await app.inject({
+        method: 'PUT',
+        url: `/usercompanies/${userCompanyId}`,
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(403);
+    });
+
+    it('should return a 403 if user Company doesnt exist', async () => {
+      const userCompanyId = new ObjectId();
+      const payload = { endDate: '2022-12-25T22:59:59.999Z' };
+
+      const res = await app.inject({
+        method: 'PUT',
+        url: `/usercompanies/${userCompanyId}`,
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(res.statusCode).toBe(403);
+    });
+
+    it('should return a 403 if user is from an other company', async () => {
```

Comment:
cf. le test analogue avec le role coach.
Ce test retourne KO a cause du fait que `authCompany` n'est pas dans DETACHMENT_ALLOWED_COMPANY_IDS

Ce titre est donc mauvais : L'appartenance ou pas a la structure n'a pas d'importance coté vendeur.

Le titre devrait etre "should return 403 if company is not allowed to be detached from" ou qqc dans le genre